src/workerd/io/worker-entrypoint.c++

```diff
@@ -568,11 +589,15 @@ kj::Promise<WorkerInterface::AlarmResult> WorkerEntrypoint::runAlarmImpl(
   KJ_IF_SOME(t, incomingRequest->getWorkerTracer()) {
     t.setEventInfo(context.now(), tracing::AlarmEventInfo(scheduledTime));
   }
-  context.getMetrics().reportTailEvent(context, [&] {
+  context.getMetrics().reportTailEvent(context.getInvocationSpanContext(), [&] {
     return tracing::Onset(
         tracing::AlarmEventInfo(scheduledTime), tracing::Onset::WorkerInfo{}, kj::none);
   });
 
+  // TODO(streaming-tail): Ensure that lifetime of outcomeObserver is handled correctly.
```

Comment:
Feels like it's easy to get outcome lifetimes wrong, so those should be reviewed closely.